### ISSUE TITLE
[v4] light-dark is broken in optimized build

### ISSUE THREAD

**Issue by imjeehoo** [tailwindlabs/tailwindcss#15438]
**What version of Tailwind CSS are you using?**

4.0.0-beta.8

**What build tool (or framework if it abstracts the build tool) are you using?**

Next.js 15.1.1, postcss 8.4.49

**What version of Node.js are you using?**

Bun 1.1.40

**What browser are you using?**

Safari

**What operating system are you using?**

macOS 15.2

**Describe your issue**

<details><summary>Original description</summary>
<div>
I defined my own color: `@theme { --color-action-sheet-cancel: light-dark(white, #2c2c2e); }` 

Color works fine in dev, but broken in production

`bun run dev`
<img width="408" alt="image" src="https://github.com/user-attachments/assets/48a5fa9b-0dd1-489a-8e39-f638c4042be2" />

`bun run build && bun run start`
<img width="437" alt="image" src="https://github.com/user-attachments/assets/1669424f-7a78-4f93-b83d-42906d89651f" />
</div>
</details> 

With optimization enabled, LightningCSS exclusively respects the color-scheme declaration set on the html element, not elsewhere.

```jsx
<html className="scheme-light dark:scheme-dark">
  <...>
    <div className="text-foreground">Hello</div>
    <div className="text-foreground scheme-only-dark">Hello</div>
  </...>
</html>
```

Without optimization:
![image](https://github.com/user-attachments/assets/2e860926-b545-4701-a25e-661290082ef9)

With optimization:
![image](https://github.com/user-attachments/assets/19b5e5cb-16be-4e86-af8e-83a58986dfa9)

**Comment by imjeehoo** [comment]
The workaround is to explicitly disable `optimize` option in production by updating `postcss.config.mjs`:
```
/** @type {import('postcss-load-config').Config} */
const config = {
  plugins: {
    '@tailwindcss/postcss': {
      optimize: false,
    },
  },
};

export default config;
```

**Comment by adamwathan** [comment]
Hey! I'm guessing this is because you are missing a `color-scheme` declaration in your CSS somewhere:

https://lightningcss.dev/transpilation.html#light-dark()-color-function

Are you using the `<meta>` tag instead? Maybe we should configure Lightning to just not transpile `light-dark` at all.

**Comment by imjeehoo** [comment]
> Hey! I'm guessing this is because you are missing a `color-scheme` declaration in your CSS somewhere:
> 
> https://lightningcss.dev/transpilation.html#light-dark()-color-function
> 
> Are you using the `<meta>` tag instead? Maybe we should configure Lightning to just not transpile `light-dark` at all.

Yes, I omitted the `color-scheme` declaration in my CSS, as I was setting it on specific elements using Tailwind `scheme-dark dark:scheme-light` and JavaScript.  

After experimenting, I discovered that with optimization enabled, the browser exclusively respects the `color-scheme` declaration set on the `html` element, not elsewhere.

```jsx
<html className="scheme-light dark:scheme-dark">
  <...>
    <div className="text-foreground">Hello</div>
    <div className="text-foreground scheme-only-dark">Hello</div>
  </...>
</html>
```

Without optimization:
<img width="899" alt="image" src="https://github.com/user-attachments/assets/c5d7a660-7fbc-4857-87f9-00de73207217" />

With optimization:
<img width="900" alt="image" src="https://github.com/user-attachments/assets/7c4459cc-79c4-4f31-8e55-7c0cbd56ae68" />

**Comment by imjeehoo** [comment]
Following is an example code based on [light-dark MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark).

Due to how it transpiles `light-dark`, we cannot put `light-dark` in variable and use it later in optimized build.

```css
:root {
  --color-background: light-dark(white, black);
  --color-foreground: light-dark(black, white);
}

* {
  background-color: var(--color-background); /* not working in optimized build */
  color: var(--color-foreground); /* not working in optimized build */

  background-color: light-dark(white, black); /* working */
  color: light-dark(black, white); /* working */
}
```

```jsx
<h1>
  <code>light-dark()</code> CSS function
</h1>
<section>
  <h2>Automatic</h2>
  <p>
    This section will react to the users system or user agent setting.
  </p>
</section>
<section className="scheme-light">
  <h2>Light</h2>
  <p>
    This section will be light due to the{" "}
    <code>color-scheme: light;</code>.
  </p>
</section>
<section className="scheme-dark">
  <h2>Dark</h2>
  <p>
    This section will be dark due to the <code>color-scheme: dark;</code>.
  </p>
</section>
```

**Comment by danie-ramdhani** [comment]
Is this same case with https://github.com/tailwindlabs/tailwindcss/issues/15018 ?

**Comment by imjeehoo** [comment]
> Is this same case with https://github.com/tailwindlabs/tailwindcss/issues/15018 ?

The issue seems to be that the `light-dark` setting in `@theme` relies solely on the `color-scheme` property of the `<html>` tag.

I’m not a CSS expert, so I’m unsure if the standard allows omitting color styles when `color-scheme` isn’t set, although most browsers appear to default to the system’s preferred color scheme. If the standard doesn’t allow it, this should be addressed.

Pesonally, I find it frustrating that Lightning CSS introduces random caveats that cause differences between development and production. Since I use Tailwind, I don’t expect to deal with its dependencies’ caveats.

I want it to handle things correctly, behave exactly as specified by the supported version of the standard, and remain consistent across development and production environments.

**Comment by imjeehoo** [comment]
> Yeah i know it's like lightningcss behaviour or something, but opening issue in in tailwindcss doesn't do anything. Try do it in lightningcss repo.

If Tailwind doesn’t consider it their bug (which I think is understandable) because the transpilation issues are entirely derived from Lightning CSS, I’m willing to open an issue there.

However, since @adamwathan mentioned he can configure it not to transpile `light-dark`, I’m waiting to hear the Tailwind team’s perspective

**Comment by adamwathan** [comment]
> Yeah i know it's like lightningcss behaviour or something, but opening issue in in tailwindcss doesn't do anything. Try do it in lightningcss repo.

Nah definitely our problem, working on a solution already.

**Comment by adamwathan** [comment]
Update — Lightning is going to expose an option for us to disable transpiling `light-dark`, has landed in master so will be able to close this one once Lightning tags a release and we can use that flag 👍 

https://github.com/parcel-bundler/lightningcss/commit/304389600fb655c5b1af55f1946518e9b755b860

**Comment by imjeehoo** [comment]
Thank you for keeping Tailwind best, and most correct framework!
